lib/core-integration/src/Test/Integration/Scenario/API/Transactions.hs

```diff
@@ -1025,26 +1043,338 @@ spec = do
+            Default Empty
+
+        -- Outgoing tx inputs and outpus size
+        let outsOut = getFromResponseList 0 outputs r
+        let insOut = getFromResponseList 0 inputs r
+        length (outsOut :: [AddressAmount t]) `shouldBe` 2
+        length (insOut :: [ApiTxInput t]) `shouldBe` 1
+
+        -- Incoming tx inputs and outpus size
+        let outsIn = getFromResponseList 1 outputs r
+        let insIn = getFromResponseList 1 inputs r
+        length (outsIn :: [AddressAmount t]) `shouldBe` 1000
+        length (insIn :: [ApiTxInput t]) `shouldBe` 1
+
+        verify r
+            [ expectResponseCode @IO HTTP.status200
+            , expectListSizeEqual 2
+            , expectListItemFieldEqual 0 direction Outgoing
+            , expectListItemFieldBetween 0 amount (feeMin + 1, feeMax + 1)
+            , expectListItemFieldEqual 0 status InLedger
+            , expectListItemFieldEqual 1 direction Incoming
+            , expectListItemFieldEqual 1 amount 1_000_000_000_000
+            , expectListItemFieldEqual 1 status InLedger
+            ]
+
+    -- This scenario covers the following matrix of cases. Cases where generated
+    -- using one of parwise test cases generation tools available online.
+    -- +----+----------+----------+------------+--------------+
+    --     |  start   |   end    |   order    |    result    |
+    -- +----+----------+----------+------------+--------------+
+    --   1 | edge     | edge     | ascending  | 2 ascending  |
+    --   2 | edge     | edge + 1 | descending | 2 descending |
+    --   3 | edge     | edge - 1 | empty      | 1st one      |
+    --   4 | edge     | empty    | empty      | 2 descending |
+    --   5 | edge + 1 | edge + 1 | empty      | 2nd one      |
+    --   6 | edge + 1 | edge - 1 | empty      | none         |
+    --   7 | edge + 1 | empty    | ascending  | 2nd one      |
+    --   8 | edge + 1 | edge     | descending | 2nd one      |
+    --   9 | edge - 1 | edge - 1 | ascending  | 1st one      |
+    --  10 | edge - 1 | empty    | descending | 2 descending |
+    --  11 | edge - 1 | edge     | empty      | 2 descending |
+    --  12 | edge - 1 | edge + 1 | empty      | 2 descending |
+    --  13 | empty    | empty    | empty      | 2 descending |
+    --  14 | empty    | edge     | empty      | 2 descending |
+    --  15 | empty    | edge + 1 | ascending  | 2 ascending  |
+    --  16 | empty    | edge - 1 | descending | 1st one      |
+    -- +----+----------+----------+------------+--------------+
+    it "TRANS_LIST_02,03 - Can limit/order results with start, end and order" $ \ctx -> do
+        (_, w, [(t1, a1), (t2, a2)]) <- fixtureWalletManyTxs ctx [10,20]
+
+        let toParam = toQueryParam . utcIso8601ToText
+        let matrix =
+                [
+                -- 1
+                  ( "?start="
+                    <> (toParam t1)
+                    <> "&end="
+                    <> (toParam t2)
+                    <> "&order=ascending"
```

Comment:
Perhaps the `toQueryString` function would be helpful here (and in other places below)?

https://github.com/input-output-hk/cardano-wallet/pull/613/files#diff-213a7e10be74fd4570beffc3464fd728R825